src/visitors/ispc_rename_visitor.hpp

```diff
@@ -0,0 +1,79 @@
+/*************************************************************************
+ * Copyright (C) 2018-2020 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#pragma once
+
+/**
+ * \file
+ * \brief \copybrief nmodl::visitor::IspcRenameVisitor
+ */
+
+#include <regex>
+#include <string>
+#include <unordered_map>
+
+#include "visitors/ast_visitor.hpp"
+
+
+namespace nmodl {
+namespace visitor {
+
+/**
+ * @addtogroup visitor_classes
+ * @{
+ */
+
+/**
+ * \class IspcRenameVisitor
+ * \brief Rename variable names to match with ISPC backend requirement
+ *
+ * Here are some examples of double constants that
+ * ISPC compiler parses: 3.14d, 31.4d-1, 1.d, 1.0d,
+ * 1d-2. Rename variables based on regex that matches
+ * those presentations.
+ *
+ */
+class IspcRenameVisitor: public AstVisitor {
+  private:
+    /// ast::Ast* node
+    std::shared_ptr<ast::Program> ast;
+
+    /// regex that matches double constant expressions
+    const std::regex double_regex = std::move(
+        std::regex("([0-9\\.]*d[\\-0-9]+)|([0-9\\.]+d[\\-0-9]*)"));
+
+    /// new name
+    const std::string new_var_name_prefix = "var_";
+
+    /// Map that keeps the renamed variables to keep the same random suffix when a variable is
+    /// renamed accross the whole file
+    std::unordered_map<std::string, std::string> renamed_variables;
```

Comment:
Do you need it ordered? What about an unordered_map?

Reply:
It's actually an unordered map but maybe I should also state it in the comment 👍

Reply:
wow, I am blind. Sorry